pyccel/stdlib/STC_Extensions/List_extensions.h

```diff
@@ -19,8 +15,34 @@ static inline i_key _c_MEMB(_pull_elem)(i_type* self, intptr_t pop_idx) {
     }
     return *(itr.ref); // Return the element that is being popped.
 }
+#if defined(_Imaginary_I) || defined(_Complex_I) || defined(_Complex)
+    /* Do nothing */
+#else
```

Comment:
```suggestion
#if !defined(_Imaginary_I) && !defined(_Complex_I) && !defined(_Complex)
```
?
I don't know if you need all of these though

Reply:
Probably not! Just to be sure. Can you think of something better?

Reply:
For lists maybe you should use `#if i_use_cmp`?

Reply:
I used it before, but to keep consistent with `sets` I opted to check for a complex `i_key`